scripts/player/ProtagonistController.gd

```diff
@@ -0,0 +1,48 @@
+extends CharacterBody2D
+
+@export var acceleration = 1500  # Acceleration in pixels per second squared
+@export var max_speed = 300     # Maximum speed in pixels per second
+@export var friction = INF     # Deceleration when not moving in pixels per second squared= 300;
+
+func _physics_process(delta: float) -> void:
+	# Get input direction
+	var direction = Vector2.ZERO
+	direction.x = Input.get_axis("ui_left", "ui_right")
+	direction.y = Input.get_axis("ui_up", "ui_down")
+
+ # Normalize direction to prevent faster diagonal movement
+	if direction.length() > 0:
+		direction = direction.normalized()
+	
+	# Apply acceleration
+	if direction != Vector2.ZERO:
+		# Accelerate in input direction, scaled by delta time
+		velocity += direction * acceleration * delta
+		# Clamp to max speed
+		velocity = velocity.limit_length(max_speed)
+	else:
+		# Apply friction when no input
+		var friction_amount = friction * delta
```

Comment:
How does the script handle the friction here?
It being set to INF brings me to think, it is unnecessary to compute friction_amount or to check whether velocity.length() > friction_amount since I expect INF to always be greater.

It could be that considering delta time is indeed required and multiplying the friction by the delta time could generate values smaller than velocity.length, and the whole point is to smoothly stop the character when you're no longer moving it. It just seems counterintuitive to me